src/operation/mod.rs

```diff
@@ -67,12 +68,17 @@ pub(crate) use insert::Insert;
 pub(crate) use list_collections::ListCollections;
 pub(crate) use list_databases::ListDatabases;
 pub(crate) use list_indexes::ListIndexes;
+#[cfg(feature = "csfle")]
+pub(crate) use raw_output::RawOutput;
 pub(crate) use run_command::RunCommand;
 pub(crate) use update::Update;
 
 const SERVER_4_2_0_WIRE_VERSION: i32 = 8;
 
 /// A trait modeling the behavior of a server side operation.
+///
+/// No methods in this trait should have default behaviors to ensure that wrapper operations
+/// replicate all behavior.  Default behavior is provided by the `OperationDefault` trait.
 pub(crate) trait Operation {
```

Comment:
The `Trait` / `TraitWithDefaults` pattern introduced here solves a gripe I've had for a while, which is that when implementing something that delegates most trait logic (e.g. `raw_output::RawOutput`) you won't get any help from the compiler if new trait methods are added that have defaults, which in turn could cause nasty logic bugs downstream.  It's definitely a little awkward and not work using for most traits, but `Operation` is critical enough that it seems worth using here.